.github/workflows/ci.yaml

```diff
@@ -29,6 +29,8 @@ jobs:
         run: poetry install --no-interaction --no-root
       - name: Run black
         run: poetry run black --check .
+      - name: Run mypy
+        run: poetry run mypy src
```

Comment:
How about type checking the tests?

Reply:
From experience, I haven't gotten too much value of type checking tests.

I have nothing against it and cannot make a strong argument against it either.

I'd suggest to restrict this PR to type checking src and you could play around with typing the checks to see if it's something you'd like.

Reply:
the idea behind static type checking is to enforce certain invariants across all possible program states. function foo always returns a list of strings, things like that. unit tests are for checking behaviour on specific inputs. they're complementary.

you don't need to typecheck your tests because they're already only running through a specifically enumerated set of inputs: the test cases. so you don't need that level of generality. furthermore, tests frequently benefit from dynamic features like monkeypatching, so they don't mesh well with type checkers.

so I agree that there's little sense in running mypy on the tests directory.

Reply:
Thanks for the info, @ckp95! I appreciate it, and I agree it makes sense to keep the typechecking only on the main source code.